src/docs/release/breaking-changes/rendereditable-stop-painting-invalid-selection.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Change RenderEditable to stop painting the caret when its selection is invalid
+description: RenderEditable no longer paints the caret when its selection is invalid.
+---
+
+## Summary
+
+When a text field's selection is `TextSelection.collapsed(offset: -1)`, or 
+a different range that contains negative positions, `RenderEditable` no 
+longer paints the caret, nor will its `onCaretChanged` callback be called.
+
+## Context
+
+The caret indicates the insertion point within the current text in an 
+active input field. Typically, when a new character is entered, the 
+caret stays immediately after it. In Flutter the caret position is 
+represented by a collapsed selection. An invalid selection is a selection
+that contains negative text positions, such as `(-1, -1)`. When the 
+selection is invalid, usually the user won't be able to modify or add text 
```

Comment:
same here how would user change the selection to a valid value.

Reply:
this feels like a bug. This meant textfield is focused but unmodifiable?

Reply:
IMEs will still be able to modify the text. IMO not allowing insertion/deletion makes the most sense when the selection is invalid, but I can't find a single use case where selection should be set to invalid (despite both iOS and Android text fields allow selection removal).